components/automate-gateway/handler/iam/v2beta/policy/policy.go

```diff
@@ -424,6 +430,62 @@ func (p *Server) ListProjects(
 	return &pb_resp.ListProjectsResp{Projects: projects}, nil
 }
 
+// IntrospectAllProjects returns the list of all projects for display in the
+// global projects filter, so it may include unassigned.
+func (p *Server) IntrospectAllProjects(
+	ctx context.Context, req *pb_req.ListProjectsReq) (*pb_resp.ListProjectsResp, error) {
+
+	projectIDs, err := p.getAllowedProjects(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// To correlate names with project IDs, use the standard ListProjects
+	// with the special flag to include "(unassigned)" too.
+	resp, err := p.projects.ListProjects(ctx, &authz.ListProjectsReq{GlobalFilterView: true})
```

Comment:
I don't feel that ListProjects needs to change here. I can't think of another scenario when ListProjects would need to return the `unassigned` project, so it seems like this is leaking concerns between two APIs.
Instead of tying ListProjects to IntrospectAllProjects, could we handle this all within the IntrospectAllProjects API? I'm thinking of a helper that checks projectIDs for `unassigned` and if found adds an `unassigned` Project to the list of projects.

Reply:
I agree that sounds nicer, @bcmdarroch , except it means that IntrospectAllProjects needs a further call to `GetProject("(unassigned)")` to find out its name. But, on balance, that sounds reasonable.

Reply:
Isn't the `unassigned` name a constant somewhere? If not can we make it one?